docs/developers/guides/token.md

```diff
@@ -0,0 +1,1287 @@
+
+    expect(res.value).toBe("[token name]");
+  });
+```
+On line 31, replace `[token name]` with the same token name that was entered in the `assembly/Token.ts`
+
+```ts linenums="34" title="assembly/__tests__/Token.spec.ts"
+  it("should get the symbol", () => {
+    const tkn = new Token();
+
+    const args = new token.symbol_arguments();
+    const res = tkn.symbol(args);
+
+    expect(res.value).toBe("[token symbol]");
+  });
+```
+On line 40, replace `[token symbol]` with the same symbol that was entered in the `assembly/Token.ts`
+
+---
+## Compiling the contract
+Once you've made all the necessary modifications and your tests are passing, you're ready to build the release version of your contract. Let's run the build script for release.
+
+```sh
+yarn build:release
+```
+
+```{ .txt .no-copy }
+Generating ABI file...
+ yarn protoc --plugin=protoc-gen-abi=./node_modules/.bin/koinos-abi-proto-gen --abi_out=abi/ assembly/proto/token.proto
+yarn run v1.22.19
+warning ../../../../../package.json: No license field
+$ /Users/ron/devstuff/projects/tmp/token-tutorial/token/node_modules/.bin/protoc --plugin=protoc-gen-abi=./node_modules/.bin/koinos-abi-proto-gen --abi_out=abi/ assembly/proto/token.proto
+installing estraverse@^5.1.0
+Done in 2.26s.
+Generating proto files...
+yarn protoc --plugin=protoc-gen-as=./node_modules/.bin/as-proto-gen --as_out=. assembly/proto/*.proto
+yarn run v1.22.19
+warning ../../../../../package.json: No license field
+$ /Users/ron/devstuff/projects/tmp/token-tutorial/token/node_modules/.bin/protoc --plugin=protoc-gen-as=./node_modules/.bin/as-proto-gen --as_out=. assembly/proto/token.proto
+Done in 0.46s.
+Generating boilerplate.ts and index.ts files...
+yarn protoc --plugin=protoc-gen-as=./node_modules/.bin/koinos-as-gen --as_out=assembly/ assembly/proto/token.proto
+yarn run v1.22.19
+warning ../../../../../package.json: No license field
+$ /Users/ron/devstuff/projects/tmp/token-tutorial/token/node_modules/.bin/protoc --plugin=protoc-gen-as=./node_modules/.bin/koinos-as-gen --as_out=assembly/ assembly/proto/token.proto
+Done in 0.38s.
+Compiling index.ts...
+node ./node_modules/assemblyscript/bin/asc assembly/index.ts --target release --use abort= --use BUILD_FOR_TESTING=0 --disable sign-extension --config asconfig.json
+```
+
+Let's run our test one more time to make sure we don't get errors.
+
+```sh
+yarn test
+```
+
+If all the tests are green, we are ready to locate the `.wasm` and `.abi` files:
+
+- `.wasm` file: `./build/release/contract.wasm`
+- `.abi` file: `./abi/token.abi`
+
+With these files, we can upload our contract to the blockchain by following the [Deploying a contract](../deploy-contract.md) documentation.
+
+## Minting the token
+
+Now that our contract is deployed to the testnet let's register our contract by issuing the `register` command `register <name:contract-name> <address:address> [abi-filename:file]`. Note that we are registering the address that we uploaded to (see above). 
```

Comment:
Do we want to just use `register_token` instead of `register`. That way the precision and stuff is done for us?

Reply:
I'm having a bit of trouble with this. When I use `register_token` I don't get the same options as when using `register`. Particularly I don't get the `mint` option, so I'm not sure how to mint the tokens.

Reply:
When we use `register_token` we bypass loading the API therefore the CLI does not provide the `mint` function and as per of this guide we are minting tokens. It's worth mentioning the `register_token` but we also need to `register` the contract normally.

Reply:
Let's walk through both `register` and `register_token` and explain the differences. Specifically that `register_token` accounts for precision but does not have mint. `register` does not account for precision but exposes all functions in the ABI.

Reply:
Would this be sufficient? 

> It's important to note that we might have been tempted to use `register_token` instead. While `register_token` accounts for precision it does not have a mint function. The `register` command does not account for precision but exposes all functions of the ABI.